File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/resources/BrukerRegistreringWrapper.java
package no.nav.fo.veilarbregistrering.registrering.bruker.resources;

import no.nav.fo.veilarbregistrering.registrering.BrukerRegistreringType;
import no.nav.fo.veilarbregistrering.registrering.bruker.BrukerRegistrering;

public class BrukerRegistreringWrapper {

    private final BrukerRegistreringType type;
    private final BrukerRegistrering registrering;

    public BrukerRegistreringWrapper(BrukerRegistrering registrering){
        this.registrering = registrering;
        type = registrering.hentType();
    }

    public BrukerRegistreringType getType() {
        return this.type;
    }

    public BrukerRegistrering getRegistrering() {
        return this.registrering;
    }

}

File: src/main/java/no/nav/fo/veilarbregistrering/registrering/bruker/resources/BrukerRegistreringWrapper.kt
package no.nav.fo.veilarbregistrering.registrering.bruker.resources

import com.fasterxml.jackson.annotation.JsonProperty
import no.nav.fo.veilarbregistrering.registrering.bruker.BrukerRegistrering
import no.nav.fo.veilarbregistrering.registrering.BrukerRegistreringType

data class BrukerRegistreringWrapper(val registrering: BrukerRegistrering) {
    //@JsonProperty(value = "type", access = JsonProperty.Access.READ_ONLY)
    val type: BrukerRegistreringType = registrering.hentType()

}